datacamp_python_project_spec_lefthandedness.ipynb

```diff
@@ -0,0 +1,255 @@
+      "text/plain": [
+       "<matplotlib.figure.Figure at 0x7ffbb88f7588>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "# load the left-handedness data and create a scatter plot\n",
+    "\n",
+    "import pandas as pd\n",
+    "from matplotlib import pyplot as plt\n",
+    "\n",
+    "data_url = \"https://gist.githubusercontent.com/mbonsma/8da0990b71ba9a09f7de395574e54df1/raw/aec88b30af87fad8d45da7e774223f91dad09e88/lh_data.csv\"\n",
+    "lefthanded_data = pd.read_csv(data_url, sep = ',')\n",
+    "\n",
+    "# this makes the plot appear in the notebook\n",
+    "%matplotlib inline\n",
+    "\n",
+    "plt.figure()\n",
+    "ax = plt.gca()\n",
+    "ax.plot(lefthanded_data['Age'], lefthanded_data['Female'], label=\"Female\", marker = 'o')\n",
+    "ax.plot(lefthanded_data['Age'], lefthanded_data['Male'], label = \"Male\", marker = 'x')\n",
+    "\n",
+    "ax.legend(loc = \"best\")\n",
```

Comment:
Maybe unnecessary to specify `loc` in this case since 'best' is the same as the default 'upper right' in this plot

Reply:
'best' is the default in 2.0+.

Reply:
@QuLogic That's what I was going to comment on at first, but then I double checked the [docstring](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/legend.py#L114), which says that `'upper right'` is the default. If I understand [the legend code](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/legend.py#L483-L486) correctly, it does look like `'upper right'` is indeed the "default", unless the rcParam for the legend loc is `'best'`, which happens to be the default for that rcParam so `loc` will be `'best'` unless there is a user override.

First I was excited to provide a PR suggesting and update to the docstring, but I  guess it seems to be correct, just that there is another default overriding the legend default, which is quite confusing... Am I understanding this correctly? I might open an issue about it then.

Reply:
The docstring is used for both Axes and Figure legends, so in a sense it's 'correct', except the Axes legend default is overridden by another default. The docstring is also still wrong in that sense, so there's always room for improvement/a PR.